api/app/db/crud.py

```diff
@@ -8,13 +8,66 @@
 from app.schemas import StationCodeList
 from app.db.models import (
     ProcessedModelRunUrl, PredictionModel, PredictionModelRunTimestamp, PredictionModelGridSubset,
-    ModelRunGridSubsetPrediction, NoonForecast, HourlyActual)
+    ModelRunGridSubsetPrediction, NoonForecast, HourlyActual, WeatherStationModelPrediction)
 import app.time_utils as time_utils
 
 logger = logging.getLogger(__name__)
 
 LATLON_15X_15 = 'latlon.15x.15'
 
+# --------------  COMMON UTILITY FUNCTIONS ---------------------------
```

Comment:
This module is starting to get a bit busy - I won't complain if someone starts splitting it up....